File: Strings/Java/CountVowelsConsonantsDigitsSpaces.java

public class CountVowelsConsonantsDigitsSpaces {
    public static void main(String[] args) {
        String str = "Hello World 123";
        int vowels = 0, consonants = 0, digits = 0, spaces = 0;
        for (char ch : str.toCharArray()) {
            if (Character.isDigit(ch)) digits++;
            else if (Character.isWhitespace(ch)) spaces++;
            else if ("aeiouAEIOU".indexOf(ch) != -1) vowels++;
            else if (Character.isLetter(ch)) consonants++;
        }
        System.out.println("Vowels: " + vowels);
        System.out.println("Consonants: " + consonants);
        System.out.println("Digits: " + digits);
        System.out.println("Spaces: " + spaces);
    }
}

File: Strings/Python/count_vowels_consonants_digits_spaces.py
from collections import Counter

def count_all(s):
    c = Counter()

    for ch in s:
        if ch.isdigit():
            c['digits'] += 1
        elif ch.isspace():
            c['spaces'] += 1
        elif ch.lower() in 'aeiouAEIOU':
            c['vowels'] += 1
        elif ch.isalpha():
            c['consonants'] += 1

    print(f"Vowels: {c['vowels']}")
    print(f"Consonants: {c['consonants']}")
    print(f"Digits: {c['digits']}")
    print(f"Spaces: {c['spaces']}")

# Example
count_all("Hello World 123")
